src/2024h2/min_generic_const_arguments.md

```diff
@@ -0,0 +1,83 @@
+
+### The next few steps
+
+In no particular order:
+- Author a document outlining everything wrong with `generic_const_exprs` as a sort of "design retrospective"
+- Implement a `min_generic_const_args` that would be ready for stabilization pending an accepted RFC (and potentially the new trait solver being stable)
+- Remove all hacks in the compiler relating to `generic_const_exprs`
+- Write an RFC for `min_generic_const_args`'s design once we know for certain it is a coherent design
+
+### The "shiny future" we are working towards
+
+The larger goal here is to lift most of the restrictions that const generics currently have:
+- Arbitrary types can be used in const generics instead of just: integers, floats, bool and char.
+    - implemented under `feature(adt_const_params)` and is relatively close to stabilization
+- Generic parameters are allowed to be used in const generic arguments.
+- Users can specify `_` as the argument to a const generic, allowing inferring the value just like with types.
+    - implemented under `feature(generic_arg_infer)` and is relatively close to stabilization
+- Associated const items can introduce generic parameters to bring feature parity with type aliases
+    - implemented under `feature(generic_const_items)`, needs a bit of work to finish it. Becomes significantly more important *after* implementing `min_generic_const_args`
+- Introduce associated const equality bounds, e.g. `T: Trait<ASSOC = N>` to bring feature parity with associated types
+    - implemented under `feature(associated_const_equality)`, blocked on allowing generic parameters in const generic arguments
+
+
+Allowing generic parameters to be used in const generic arguments is the only part of const generics that requires significant amounts of work while also having significant benefit. Everything else is already relatively close to the point of stabilization. I chose to specify this goal to be for implementing `min_generic_const_args` over "stabilize the easy stuff" as I would like to know whether the implementation of `min_generic_const_args` will surface constraints on the other features that may not be possible to easily fix in a backwards compatible manner. Regardless I expect these features will still progress while `min_generic_const_args` is being implemented.
+
+
+## Design axioms
+
+- Do not block future extensions to const generics
+- It should not feel worse to write type system logic with const generics compared to type generics
+- Avoid post-monomorphization errors
+- The "minimal" subset should not feel arbitrary
+
+## Ownership and other resources
+
+**Owner:** BoxyUwU, project-const-generics lead
+
+### Support needed from the project
+
+- Lang
+    - Potential review of a `min_generic_const_args` RFC
+    - Occasional bikeshedding
```

Comment:
I'm a bit concerned about review/implementation capacity for this. I know that *I* have a lot of time to put into this, but every PR needs 2 people, one to implement and one to review lol. Currently @camelid, you're working on implementation for this but I have no idea how much you want to commit to this and if you have the available time.

Reply:
@compiler-errors has mentioned being okay with helping out with reviews and implementation so I think this is not going to be a problem :-)